common/djangoapps/third_party_auth/models.py

```diff
@@ -293,6 +299,12 @@ def match_social_auth(self, social_auth):
         prefix = self.idp_slug + ":"
         return self.backend_name == social_auth.provider and social_auth.uid.startswith(prefix)
 
+    def get_remote_id_from_social_auth(self, social_auth):
+        """ Given a UserSocialAuth object, return the remote ID used by this provider. """
+        assert self.match_social_auth(social_auth)
+        # Remove the prefix from the UID
+        return social_auth.uid[len(self.idp_slug) + 1:]
```

Comment:
@cpennington The prefix is an internal implementation detail, and API users (especially external ones like a university) should not have to deal with it. The API results include pairs of (`provider_id`, `remote_id`), and even though the remote id is now unprefixed, the `provider_id` takes the place of the prefix so those pairs are always going to be unique.